edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/DeviceProxy.cs

```diff
@@ -40,6 +40,16 @@ public class DeviceProxy : IDeviceProxy
             this.isActive = new AtomicBoolean(true);
             this.isDirectClient = isDirectClient;
 
+            // when a child edge device connects, it uses $edgeHub identity.
+            // Although it is a direct client, it uses the indirect topics
+            if (identity is ModuleIdentity moduleIdentity)
+            {
+                if (string.Equals(moduleIdentity.ModuleId, "$edgeHub"))
```

Comment:
you can use Constants.EdgeHubModuleId from  Microsoft.Azure.Devices.Edge.Hub.Core